lib/oli/delivery/activity_provider.ex

```diff
@@ -241,22 +254,30 @@ defmodule Oli.Delivery.ActivityProvider do
   end
 
   # fulfill any resource group types
-  defp do_fulfill(fulfillment_state, %{"type" => type} = model_component, group_id, survey_id) do
-    if PageContent.is_resource_group?(model_component) do
+  defp do_fulfill(
+         fulfillment_state,
+         %{"type" => type} = model_component,
+         group_id,
+         survey_id,
+         user,
+         section_slug
+       ) do
+    if PageContent.is_resource_group?(model_component) &&
+         Audience.is_intended_audience?(model_component["audience"], user, section_slug, false) do
```

Comment:
So I think this is likely the wrong place to do this filtering, as evidenced by the bug that appears when one places a survey with an activity in an instructor only group.

I think we must add an upfront, earlier step that prunes out the page content to remove audience groups that do not match.  Then, that "transformed" page content must be the content that is stored in the `resource_attempt` record in the `:content` field.  The principal reason for this `:content` value is that it stores the transformed page content after activity realization has occurred - where activity bank selections are turned into actual activity-references.  The results of audience filtering must be captured in this field.